src/audio/dai.c

```diff
@@ -646,6 +646,11 @@ static int dai_position(struct comp_dev *dev, struct sof_ipc_stream_posn *posn)
 static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 {
 	struct dai_data *dd = comp_get_drvdata(dev);
+	struct sof_ipc_comp_dai *dai = (struct sof_ipc_comp_dai *)&dev->comp;
+
+	/* return if config is not for this comp */
+	if (dai->dai_index != config->dai_index)
+		return 0;
```

Comment:
Add error trace and return -EINVAL?
This looks like something we'd need to do for all DAIs btw.

Reply:
@plbossart this is not an error. Basically when the dai_config ipc arrives, the ipc handler calls the dai_config() callback for all comps. So we should just ignore the request if the dai_config is not address for it.

Reply:
So either we ignore the request here or maybe fix the ipc handler to send the config only to the relevant comps. I think the latter makes more sense. I will update the PR.